src/Iris/BI/Notation.lean

```diff
@@ -31,7 +31,13 @@ partial def unpackIprop [Monad m] [MonadRef m] [MonadQuotation m] : Term → m T
   | `(?$P:ident)             => do `(?$P)
   | `(($P))                  => do `(($(← unpackIprop P)))
   | `($P $[ $Q]*)            => do ``($P $[ $Q]*)
-  | `(if $c then $t else $e) => do `(if $c then $(← unpackIprop t) else $(← unpackIprop e))
+  | `(if $c then $t else $e) => do
+    -- TODO: Why does `(if $c then $(← unpackIprop t) else $e) give
+    -- "invalid use of `(<- ...)`, must be nested inside a 'do'
+    -- expression" here, but not in the other cases?
+    let t ← unpackIprop t;
+    let e ← unpackIprop e;
+    `(if $c then $t else $e)
```

Comment:
```suggestion
  | `(if $c then $t else $e) => do
    let t ← unpackIprop t
    let e ← unpackIprop e
    `(if $c then $t else $e)
```
This is a known issue, it triggers because we are using `<-` inside an `if` statement body, which is a footgun in normal usage. It's a false positive here because the `if` is actually in a quotation so it isn't actually running, but the check which produces the error misses this issue. We hit several instances of this when the error was first rolled out in user code, and decided that it's best just to work around it for the present and do as you did here.

Reply:
I see, thanks! I've removed the comment.